ChiModules/KEigenvalueSolver/IterativeMethods/power_iteration.cc

```diff
@@ -25,161 +28,128 @@ using namespace LinearBoltzmann;
 
+  //======================================== Start power iterations
+  double F_prev = 1.0;
+  double k_eff_prev = 1.0;
+  int nit = 0;      //number of iterations
+  bool converged = false;
   while (nit < options.max_iterations)
   {
-    chi_log.Log(LOG_0VERBOSE_2)
-      << "\n********** Starting source iterations";
 
+    //============================== Clear source moments
+    q_moments_local.assign(q_moments_local.size(), 0.0);
+
+    //============================== Set the fission source
+    SetKSource(groupset, q_moments_local,
+               APPLY_AGS_FISSION_SOURCE |
+               APPLY_WGS_FISSION_SOURCE);
 
-    // ----- Start inner source iterations
-    double pw_change_prev = 1.0;
-    bool si_converged = false;
-    for (int si_nit=0; si_nit<groupset.max_iterations; si_nit++)
+    //============================== Converge the scattering source with
+    //                               a fixed fission source
+    if (groupset.iterative_method == IterativeMethod::CLASSICRICHARDSON)
     {
-      // ----- Set source and sweep
-      SetKSource(groupset, APPLY_MATERIAL_SOURCE |
-                           APPLY_AGS_SCATTER_SOURCE | APPLY_WGS_SCATTER_SOURCE |
-                           APPLY_FISSION_SOURCE);
-      groupset.angle_agg.ZeroOutgoingDelayedPsi();
-      phi_new_local.assign(phi_new_local.size(),0.0);
-      SweepScheduler.Sweep();
-
-      // ----- Compute convergence parameters
-      double pw_change = ComputePiecewiseChange(groupset);
-      ScopedCopySTLvectors(groupset, phi_new_local, phi_old_local);
-      double rho = sqrt(pw_change/pw_change_prev);
-      pw_change_prev = pw_change;
-      nit += 1;
-
-      if (si_nit==0) rho = 0.0;
-      if (pw_change<std::max(groupset.residual_tolerance*rho,1.0e-10))
-        si_converged = true;
-
-      // ----- Print iteration information
-      std::string offset = "    ";
-      std::stringstream si_iter_info;
-      si_iter_info
-        << chi_program_timer.GetTimeString() << " "
-        << offset
-        << "WGS groups ["
-        << groupset.groups.front().id
-        << "-"
-        << groupset.groups.back().id
-        << "]"
-        << " Source Iteration " << std::setw(5) << si_nit
-        << " Point-wise change " << std::setw(14) << pw_change;
-        
-      if (si_converged)
-        si_iter_info << " CONVERGED\n";
-      chi_log.Log(LOG_0VERBOSE_1) << si_iter_info.str();
-
-      if (si_converged) {
-        chi_log.Log(LOG_0VERBOSE_1)
-            << "\nSource iterations converged in "
-            << si_nit << " iteratrions.\n";
-        break;
-      }
-    }//for source iterations
-
-    if (!si_converged) {
-      chi_log.Log(LOG_ALLVERBOSE_1)
-          << "\n!!!WARNING!!! "
-             "Source iterations did not converge.\n";
+      ClassicRichardson(groupset, 0, sweep_scheduler,
+                        APPLY_WGS_SCATTER_SOURCE | APPLY_AGS_SCATTER_SOURCE,
+                        false);
```

Comment:
I think it might be good to change `false` to
```c++
options.verbose_inner_iterations
```
Same thing for the GMRES function

Reply:
I agree with this.

Reply:
Done.